modules/rest/src/main/java/eu/cloudnetservice/modules/rest/setup/HttpHostAndPortSetup.java

```diff
@@ -0,0 +1,85 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package eu.cloudnetservice.modules.rest.setup;
+
+import eu.cloudnetservice.driver.network.HostAndPort;
+import eu.cloudnetservice.modules.rest.CloudNetRestModule;
+import eu.cloudnetservice.node.Node;
+import eu.cloudnetservice.node.config.HttpHostAndPort;
+import eu.cloudnetservice.node.config.RestConfiguration;
+import eu.cloudnetservice.node.console.animation.setup.ConsoleSetupAnimation;
+import eu.cloudnetservice.node.console.animation.setup.answer.Parsers;
+import eu.cloudnetservice.node.console.animation.setup.answer.QuestionAnswerType;
+import eu.cloudnetservice.node.console.animation.setup.answer.QuestionListEntry;
+import eu.cloudnetservice.node.setup.DefaultSetup;
+import lombok.NonNull;
+
+public class HttpHostAndPortSetup implements DefaultSetup {
+
+  private final CloudNetRestModule cloudNetRestModule;
+
+  public HttpHostAndPortSetup(CloudNetRestModule cloudNetRestModule) {
+    this.cloudNetRestModule = cloudNetRestModule;
+  }
+
+  @Override
+  public void applyQuestions(@NonNull ConsoleSetupAnimation animation) {
+    animation.addEntries(
+      QuestionListEntry.<Boolean>builder()
+        .key("ignorePort")
+        .translatedQuestion("command-http-setup-question-reverseProxy")
+        .answerType(QuestionAnswerType.<Boolean>builder().parser(Parsers.bool()).addResultListener((__, result) -> {
+          if (!result) {
+            animation.addEntriesFirst(
+              QuestionListEntry.<HostAndPort>builder()
+                .key("hostAndPort")
+                .translatedQuestion("command-http-setup-question-hostAndPort")
+                .answerType(
+                  QuestionAnswerType.<HostAndPort>builder().parser(Parsers.validatedHostAndPort(true)).build())
+                .build());
+          } else {
+            animation.addEntriesFirst(
+              QuestionListEntry.<String>builder()
+                .key("domain")
+                .translatedQuestion("command-http-setup-question-domain")
+                .answerType(QuestionAnswerType.<String>builder().parser(Parsers.nonEmptyStr()).build()).build());
+          }
+        }).build()).build());
+    animation.addEntries(QuestionListEntry.<Boolean>builder()
+      .key("ssl")
+      .translatedQuestion("command-http-setup-question-ssl")
+      .answerType(QuestionAnswerType.<Boolean>builder().parser(Parsers.bool()).build()).build());
+
+
+  }
+
+  @Override
+  public void handleResults(@NonNull ConsoleSetupAnimation animation) {
```

Comment:
I don't like the ternary usages in this function. But if @derklaro likes it, it's okay.